yoti-sdk/src/main/java/com/yoti/mobile/android/sdk/YotiSDK.java

```diff
@@ -130,7 +144,74 @@ static void startScenario(final Context context, final String useCaseId,
         }
 
         YotiSDKLogger.debug("Started scenario " + useCaseId);
-        KernelSDKIntentService.startActionStartScenario(context, useCaseId, onYotiCalledResultReceiver);
+        KernelSDKIntentService.startActionStartScenario(context, useCaseId, onYotiCalledResultReceiver, getLaunchingAppScheme(sdkButtonTheme, appPackageInfo));
+    }
+
+    private static PackageInfo getAppPackageInfoBasedOnTheme(PackageManager packageManager, ButtonTheme buttonTheme) throws YotiSDKAppNotInstalledException {
+        PackageInfo packageInfo = null;
+        try {
+            switch (buttonTheme) {
+                case THEME_YOTI:
+                case THEME_PARTNERSHIP:
+                    if (checkAppInstalled(packageManager, YotiAppDefs.YOTI_APP_PACKAGE)) {
+                        packageInfo = packageManager.getPackageInfo(YotiAppDefs.YOTI_APP_PACKAGE, 0);
+                    } else if (checkAppInstalled(packageManager, YotiAppDefs.EASY_ID_APP_PACKAGE)) {
+                        packageInfo = packageManager.getPackageInfo(YotiAppDefs.EASY_ID_APP_PACKAGE, 0);
+                    } else {
+                        AppType appType = buttonTheme == THEME_YOTI ? YOTI_APP : PARTNERSHIP_APP;
+                        throw new YotiSDKAppNotInstalledException(appType, appType.getValue()+" app not installed");
+                    }
+                    break;
+                case THEME_EASYID:
+                    if (checkAppInstalled(packageManager, YotiAppDefs.EASY_ID_APP_PACKAGE)
+                            && checkEasyAppWithSchemeAvailable(packageManager)) {
```

Comment:
Why we only check the scheme only for Easy Id?

Reply:
We are launching specific app based on the scheme, with EasyId theme user can launch only EasyID app. We have added EasyID specific scheme in 3.29.0, this one is added to avoid crash if we launch app with EasyId scheme.